fluent-syntax/src/parser.js

```diff
@@ -2,7 +2,20 @@
 
 import * as AST from './ast';
 import { FTLParserStream } from './ftlstream';
-import { error, getErrorSlice } from './errors';
+import { error } from './errors';
+
+export function lineOffset(source, pos) {
+  // Substract 1 to get the offset.
+  return source.substring(0, pos).split('\n').length - 1;
+}
+
+export function columnOffset(source, pos) {
+  const lastLineBreak = source.lastIndexOf('\n', pos);
+  return lastLineBreak === -1
+    ? pos
+    // Substracting two offsets gives length; substract 1 to get the offset.
+    : pos - lastLineBreak - 1;
+}
```

Comment:
Interesting. I think we could implement it the compare-locales way, too. I think I'm going to keep `source.split` approach for now and consult with @zbraniecki about his preference, too.